packages/gatsby-plugin-feed/src/internals.js

```diff
@@ -74,3 +76,24 @@ export const defaultOptions = {
     },
   ],
 }
+
+// https://stackoverflow.com/questions/13852852/surely-es6-must-have-a-way-to-merge-two-javascript-objects-together-what-is-it
+const isObject = item => item && typeof item === 'object' && !Array.isArray(item)
+
+export function mergeDeep(target, ...sources) {
```

Comment:
Lodash is used in the core Gatsby package as well as in many of the other plugins which makes it effectively "free" for you to use here.